Revise documentation for `NetworkParameters` and related types.

jonathanknowles opened the pull request:
# Related Issues

#1693 

# Overview

PR #1700 changed the way that network parameters are defined.

This PR revises and adds documentation for the following types:

- [x] `NetworkParameters`
- [x] `GenesisParameters`
- [x] `ProtocolParameters`
- [x] `TxParameters`

jonathanknowles committed:
Use "Network Parameters" section header.

"Network Parameters" represents the complete set of parameters that are
currently in use by the network.

This set includes parameters defined by the genesis block and also protocol
parameters that can be updated through the update system.

jonathanknowles committed:
Add documentation for `GenesisParameters`.

jonathanknowles committed:
Add documentation for `ProtocolParameters`.

jonathanknowles committed:
Revise documentation for `TxParameters`.

jonathanknowles committed:
Revise documentation for `NetworkParameters`.

rvl commented on lib/core/src/Cardano/Wallet/Primitive/Types.hs:
```diff
@@ -1351,11 +1354,17 @@ slotParams gp =
 data ProtocolParameters = ProtocolParameters
     { decentralizationLevel
         :: Quantity "percent" Percentage
+        -- ^ The current level of decentralization in the network.
```
You could add that this is the _d_ parameter from the formal spec.

jonathanknowles replied:
Fixed in a417440ceb99cfc997651e6055eefc6453048470.

rvl commented on lib/core/src/Cardano/Wallet/Primitive/Types.hs:
```diff
@@ -1351,11 +1354,17 @@ slotParams gp =
+        --   * '100%' indicates that the network is /completely decentralized/.
     , txParameters
         :: TxParameters
+        -- ^ Parameters relating to transactions.
```
```suggestion
        -- ^ Parameters which affect transaction creation.
```

jonathanknowles replied:
Fixed in cfaf9ac6a561b94034465d28ad883835a13ac38b.

rvl reviewed (approved):
Better

jonathanknowles committed:
Include reference to the `d` parameter from the formal spec.

In response to review feedback:

https://github.com/input-output-hk/cardano-wallet/pull/1715#discussion_r434289213

jonathanknowles committed:
Revise comment for `txParameters` field.

In response to review feedback:

https://github.com/input-output-hk/cardano-wallet/pull/1715#discussion_r434289299

rvl commented on lib/core/src/Cardano/Wallet/Primitive/Types.hs:
```diff
@@ -1279,26 +1279,29 @@ computeUtxoStatistics btype utxos =
 
--- | Initial blockchain parameters loaded from the application configuration and
--- genesis block.
+-- | Records the complete set of parameters currently in use by the network.
```
Actually this isn't the complete set of parameters. Just those which are relevant to the wallet.

jonathanknowles replied:
Fixed in e26d3ae8bce792b5530bd8f82324accd95a0c668.

jonathanknowles committed:
Revise documentation for `NetworkParameters`.

Point out that this type only includes parameters that are relevant to
the wallet.

In response to review feedback:

https://github.com/input-output-hk/cardano-wallet/pull/1715#discussion_r434291133
